tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -55,62 +57,85 @@ def _set_default_params(self):
+            'min_samples_leaf': Int(lower=1, upper=10, default=1),
         }
         return spaces
 
-    def _fit(self, X_train, y_train, time_limit=None, **kwargs):
+    def _fit(self, X_train, y_train, time_limit=None, warm_start=True, warm_iter=None, **kwargs):
+        """
+            Additional Parameters
+            warm_start: False
+                If warm_start is True, it allows the model to perform incremental training.
+                E.g. calling model.fit(X, y) multiple times would build upon the intermediate models and
+                increase the number of trees used. See also warm_iter
+            warm_iter: None
+                This sets the humber of additional trees to add for incremental training.
+                if warm_iter is None, the number of additional trees set by the model hyperparameter, such as "n_estimators"
+       """
+        #for warm start, simply allow more trees to form
         time_start = time.time()
         max_memory_usage_ratio = self.params_aux['max_memory_usage_ratio']
         hyperparams = self.params.copy()
         n_estimators_final = hyperparams['n_estimators']
 
-        n_estimators_minimum = min(40, n_estimators_final)
-        if n_estimators_minimum < 40:
-            n_estimators_test = max(1, math.floor(n_estimators_minimum/5))
-        else:
-            n_estimators_test = 8
+        if warm_start:
+            hyperparams["warm_start"] = True
 
         X_train = self.preprocess(X_train)
-        n_estimator_increments = [n_estimators_final]
 
-        # Very rough guess to size of a single tree before training
-        if self.problem_type == MULTICLASS:
-            if self.num_classes is None:
-                num_trees_per_estimator = 10  # Guess since it wasn't passed in, could also check y_train for a better value
+        if not (self.model and warm_start):
+            n_estimators_minimum = min(40, n_estimators_final)
+            if n_estimators_minimum < 40:
+                n_estimators_test = max(1, math.floor(n_estimators_minimum/5))
             else:
-                num_trees_per_estimator = self.num_classes
-        else:
-            num_trees_per_estimator = 1
-        bytes_per_estimator = num_trees_per_estimator * len(X_train) / 60000 * 1e6  # Underestimates by 3x on ExtraTrees
-        available_mem = psutil.virtual_memory().available
-        expected_memory_usage = bytes_per_estimator * n_estimators_final / available_mem
-        expected_min_memory_usage = bytes_per_estimator * n_estimators_minimum / available_mem
-        if expected_min_memory_usage > (0.5 * max_memory_usage_ratio):  # if minimum estimated size is greater than 50% memory
-            logger.warning(f'\tWarning: Model is expected to require {round(expected_min_memory_usage * 100, 2)}% of available memory (Estimated before training)...')
-            raise NotEnoughMemoryError
-
-        if n_estimators_final > n_estimators_test * 2:
+                n_estimators_test = 8
```

Comment:
Not sure if these estimators are good picks for any dataset. Ideally this should be a function of dataset length and columns number.

Reply:
The code was refactoring the previous implementation. I didn't change any heuristics that existed before. I think any new changes to RF model should be handled by the AG team.